### PR TITLE
Add status_printf to print messages to the lcd status line.

### PR DESCRIPTION
**Pull request by thinkyhead**
Also add a hotend indicator to the heating message displayed on the lcd status line.

Tweaked #5871

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -5308,7 +5308,7 @@ inline void gcode_M104() {
       }
     #endif
 
-    if (code_value_temp_abs() > thermalManager.degHotend(target_extruder)) LCD_MESSAGEPGM(MSG_HEATING);
+    if (code_value_temp_abs() > thermalManager.degHotend(target_extruder)) status_printf(0, "E%i %s", target_extruder + 1, MSG_HEATING);
```
Comment (manianac) [inline Marlin/Marlin_main.cpp:5311]: Why not use `vsnprintf_P`?  Guessing the SRAM savings are negligible?

Comment (thinkyhead) [reply Marlin/Marlin_main.cpp:5311]: #5892

**Comment by ghost** [comment]
I am using a `REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER`.

What do I change in these to get the "Heating..." to display when the hotend is heating?
```cpp
if (thermalManager.isHeatingHotend(target_extruder))
  status_printf(1, "E%i %s", target_extruder + 1, PSTR(MSG_HEATING));
```
```cpp
void status_printf(uint8_t level, const char *status, ...) {
  if (level < lcd_status_message_level) return;
  lcd_status_message_level = level;
  va_list args;
  va_start(args, status);
  vsnprintf_P(lcd_status_message, 3 * (LCD_WIDTH), status, args);
  va_end(args);
  lcd_finishstatus(level > 0);
}
```

**Comment by thinkyhead** [comment]
> What do I change…?

Nothing should need to be done. Both `M104` and `M109` display this message automatically.

**Comment by ghost** [comment]
Ok, so... I took my own advice and downloaded RCBugFix just now. I set the minimum setup to get it to print. No extra options except for `DOGLCD` and `REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER`
(no eeprom, no SD, no filament change, no RGBs, no probe, nothing).
I uploaded it to my printer.
Status message says: `3D Printer ready.`
I send M104 S50 via Arduino Serial monitor. Status message now says: `Firmware/Marlin PROTO`
I send M104 S0. Status message now says: `3D Printer ready.`
I send M109 S50. Status message now says: `Firmware/Marlin PROTO`
Once it reached 50, status message now says: `Heating done.`
I send M104 S0. Status message now says: `3D Printer ready.`

Am I the only one with this new issue?

I am going to enable different options and try again. Because I had `ILLED.` on my build of Marlin.

**Comment by ghost** [comment]
I took the same copy of Marlin from my last post and enabled `SDSUPPORT`.
I uploaded it to my printer.
Status message says: `3D Printer ready.`
I send M104 S50. Status message says: `PROTOCOL_VERSION:1.0 `
I send M104 S0. Status message says: `3D Printer ready.`
I send M109 S50. Status message says: `PROTOCOL_VERSION:1.0 `
Once it reached 50, status message says: `Heating done.`
I send M104 S0. Status message now says: `3D Printer ready.`

----
I disabled `SDSUPPORT`.
I enabled `FIX_MOUNTED_PROBE` (Left all probe options default)
I uploaded it to my printer.
I send M104 S50. Status message says: ` Machine_TYPE:3D Prin`
I send M104 S0. Status message says: `3D Printer ready.`
I send M109 S50. Status message says: ` Machine_TYPE:3D Prin`
Once it reached 50, status message says: `Heating done.`
I send M104 S0. Status message says: `3D Printer ready.`

----
I re-enabled `SDSUPPORT` with `FIX_MOUNTED_PROBE`
I uploaded to my printer.
I send M104 S50. Status message says: `L_VERSION:1.0 MACHINE`
I send M104 S0. Status message says: `3D Printer ready.`
I send M109 S50. Status message says: `L_VERSION:1.0 MACHINE`
Once it reached 50, status message says: `Heating done.`
I send M104 S0. Status message says: `3D Printer ready.

**Comment by Tannoo** [comment]
I disabled `SDSUPPORT`.
I enabled `FIX_MOUNTED_PROBE` (Left all probe options default)
I uploaded it to my printer.
I send M104 S50. Status message says: ` Machine_TYPE:3D Prin`
I send M104 S0. Status message says: `3D Printer ready.`
I send M109 S50. Status message says: ` Machine_TYPE:3D Prin`
Once it reached 50, status message says: `Heating done.`
I send M104 S0. Status message says: `3D Printer ready.`

**Comment by Tannoo** [comment]
I re-enabled `SDSUPPORT` with `FIX_MOUNTED_PROBE`
I uploaded to my printer.
I send M104 S50. Status message says: `L_VERSION:1.0 MACHINE`
I send M104 S0. Status message says: `3D Printer ready.`
I send M109 S50. Status message says: `L_VERSION:1.0 MACHINE`
Once it reached 50, status message says: `Heating done.`
I send M104 S0. Status message says: `3D Printer ready.

**Comment by ghost** [comment]
So, I have to ask, What combination of options need to be set to get the `Heating...` message now?

**Comment by ghost** [comment]
I guess the version I use has too many options enabled, giving me the `ILLED.` status message. Because, KILLED, would be the end of the line. lol

**Comment by ghost** [comment]
I enabled `BLUETOOTH`, `SDSUPPORT`, and `FIX_MOUNTED_PROBE`
Sent M104 S50. Status message says: `L_VERSION:1.0 MACHINE`
No change than without `BLUETOOTH`.

----
I disabled `BLUETOOTH` and enabled `FILAMENT_CHANGE_FEATURE`.
So this leaves `SDSUPPORT`, and `FIX_MOUNTED_PROBE` and `FILAMENT_CHANGE_FEATURE` enabled.
Sent M104 S50. Status message says: `Marlin 1.1.0-RCBugFix`
lol

----
I diabled `FIX_MOUNTED_PROBE`.
So that leaves `SDSUPPORT` and `FILAMENT_CHANGE_FEATURE` enabled.
Sent M104 S50. Status message says: `145 S<index> out of r`

That's different.

----
I enabled `FILAMENT_RUNOUT_SENSOR`.
So, `SDSUPPORT`, `FILAMENT_CHANGE_FEATURE`, and `FILAMENT_RUNOUT_SENSOR` enabled.
Sent M104 S50. Status message says: `github.com/MarlinFirm`

Interesting.... hidden advertisement?

**Comment by Tannoo** [comment]
I disabled `BLUETOOTH and enabled `FILAMENT_CHANGE_FEATURE`
Sent M104 S50. Status message says: `Marlin 1.1.0-RCBugFix`
lol

**Comment by Tannoo** [comment]
I diabled `FIX_MOUNTED_PROBE`. So that leaves 'SDSUPPORT' and `FILAMENT_CHANGE_FEATURE` as the only options currently enabled.
Sent M104 S50. Status message says: `145 S<index> out of r`

That's different.

**Comment by Tannoo** [comment]
I enabled `FILAMENT_RUNOUT_SENSOR`.
So, `SDSUPPORT`, `FILAMENT_CHANGE_FEATURE`, and `FILAMENT_RUNOUT_SENSOR` enabled.
Sent M104 S50. Status message says: `github.com/MarlinFirm`

Interesting.... hidden advertisement?

**Comment by ghost** [comment]
Let's add `EEPROM_SETTINGS` to the mix.
Sent M104 S50. Status message says: ` ready.`

**Comment by ghost** [comment]
Let's remove `SDSUPPORT`. That will leave:
`FILAMENT_CHANGE_FEATURE`, `FILAMENT_RUNOUT_SENSOR`, and `EEPROM_SETTINGS`
Sent M104 S50. Status message says: `ithub.com/MarlinFirmw`

**Comment by ghost** [comment]
Ok, I hope that is enough to go on. Because this can go for hours. lol
Let me know if anyone wants more info.

**Comment by WheresWaldo** [comment]
I am seeing reports of spurious display text as soon as the extruder heater starts, we have a small group of people that are using RCBF on Robo3D R1's. I would say that #5944 is indeed a duplicate of the issue [Tannoo ](https://github.com/Tannoo)is seeing. I am not sure it is the same as #5894 as he mentioned RC8 and I did not see this behavior until this PR.

**Comment by thinkyhead** [comment]
> No extra options except for `DOGLCD` and `REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER`

No need to enable `DOGLCD` when using `REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER`.

I'm sure the issue here is caused by a bug in `status_printf`, or how I'm using it. I'll see about fixing that asap. (#5948)

**Pull request merged**